src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -127,7 +129,7 @@ public FormValidation doCheckCredentialsId(@AncestorInPath Item project,
             return FormValidation.ok();
         }
 
-        public FormValidation doCheckUrl(@AncestorInPath Item project,
+        public FormValidation doCheckUrl(@AncestorInPath Job project,
```

Comment:
cause `System.getEnv()` is just wrong ! It's missing all variables set by various environment providers. 
Also, please note a GitSCM can't have anything but a Job as parent, so using general purpose `Item` type here doesn't make much sense.